### PR TITLE
DeveloperGuide: Add effort

### PR DESCRIPTION
**Pull request by EmilyOng**
Added an [Effort](https://nus-cs2103-ay2223s1.github.io/website/admin/tp-deliverables.html#deliverable-developer-guide-dg) section, do help to check if it is worded appropriately and if there is anything more to add or remove, thanks!

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -805,3 +806,60 @@ testers are expected to do more *exploratory* testing.
...
+
+In addition, PennyWise features a unique tab-based layout which differs from [AddressBook Level-3](https://github.com/se-edu/addressbook-level3)'s
+single layout format. Using a tab-based layout required us to put in more thought on how and what data to render based on the
+tab that is visible. Furthermore, it required us to manipulate the tab toggling for certain commands. Nevertheless, we believe
+that adding a tab-based layout is a great addition to provide a modern touch to the application, and helps to make full use
+of the screen real-estate.
+
+Finally, implementing and styling the UI was not easy. PennyWise's UI is responsive and respects a unified colour theme and
+consistent font family. This helps to provide a standardised and modern look-and-feel to the application. Many rounds of
+iterations went into designing PennyWise, requiring us to overhaul the existing CSS file in [AddressBook Level-3](https://github.com/se-edu/addressbook-level3).
```
Comment (JasonYapzx) [inline docs/DeveloperGuide.md:857]: Very detailed, this is probably the main issue we have. Thanks for adding!

**Review by JasonYapzx — APPROVED** [review]
If the others have things to add for effort wise we can copy it into this pr and merge directly, otherwise i think this is good to merge. Thanks emily!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -805,3 +806,60 @@ testers are expected to do more *exploratory* testing.
...
+In general, PennyWise was not an easy application to develop. We have spent much time tailoring our product requirements, value
+proposition, user stories and use cases, before implementing all the features that we have planned out, as well as some nice-to-have features.
+Furthermore, there have been a great amount of work in writing test cases, enabling us to achieve a code coverage of close to
+70%.
+
+### Challenges Faced
+
+#### Models
+
+In [AddressBook Level-3](https://github.com/se-edu/addressbook-level3), there were only 1 `Person` class, which made it straightforward
```
Comment (sharmaine1028) [inline docs/DeveloperGuide.md:826]:
```suggestion
In [AddressBook Level-3](https://github.com/se-edu/addressbook-level3), there was only 1 `Person` class, which made it straightforward
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -805,3 +806,60 @@ testers are expected to do more *exploratory* testing.
...
+to define fields and actions associated with a `Person`. However, in PennyWise, we have both an `Income` class and an `Expense`
+class. Thus, all instances of `AddressBook` and its related entities had to be removed and refactored throughout the project.
+Furthermore, we needed to take great care in implementing an `Entry` class, which acts as a parent of the `Income` and `Expenditure`
```
Comment (sharmaine1028) [inline docs/DeveloperGuide.md:829]:
```suggestion
Furthermore, we needed to take great care in implementing an `Entry` class, which acts as a parent of the `Income` and `Expense`
```

**Review by sharmaine1028 — APPROVED** [review]
Very well done!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (EmilyOng) [reply docs/DeveloperGuide.md:829]: Thanks!

**Pull request merged**